github/github.go

```diff
@@ -986,6 +996,59 @@ func (g *GitHub) UpdateReleasePage(
 	return release, nil
 }
 
+// toRepositoryRelease builds a repository release from the set of options
+func (urpo *UpdateReleasePageOptions) toRepositoryRelease() *github.RepositoryRelease {
+	request := &github.RepositoryRelease{}
+
+	if urpo.Name != "" {
+		request.Name = &urpo.Name
+	}
+	if urpo.Body != "" {
+		request.Body = &urpo.Body
+	}
+
+	request.Draft = &urpo.Draft
+	request.Prerelease = &urpo.Prerelease
+
+	if urpo.Latest {
+		request.MakeLatest = ptr.To("true")
+	} else {
+		request.MakeLatest = ptr.To("false")
+	}
+
+	return request
+}
+
+// UpdateReleasePageWithOptions updates release pages (same as UpdateReleasePage),
+// but does so by taking a UpdateReleasePageOptions parameter. It will _not_ set
+// a release as latest unless the corresponding option is set.
+func (g *GitHub) UpdateReleasePageWithOptions(owner, repo string,
+	releaseID int64,
+	tag, commitish string,
+	opts *UpdateReleasePageOptions,
+) (release *github.RepositoryRelease, err error) {
+	logrus.Infof("Updating release page for %s", tag)
+
+	if opts == nil {
+		opts = &UpdateReleasePageOptions{}
+	}
+
+	releaseData := opts.toRepositoryRelease()
+	releaseData.TagName = &tag
+	releaseData.TargetCommitish = &commitish
+
+	// Call the client
+	release, err = g.Client().UpdateReleasePage(
+		context.Background(), owner, repo, releaseID, releaseData,
```

Comment:
We probably want to pass that context from the caller.

Reply:
This is not done in any method in this client so it felt weird to add this now, just for a specific method. I can do that though if you prefer it?

Reply:
Hm, agree. Let's add context support at some later point in time.

Reply:
I agree with @embik here. All other functions in this package are using `context.Background()` this way. I agree that it should be done as @saschagrunert said, but I'd leave that for some bigger/future refactoring.